app/src/main/java/org/lineageos/glimpse/viewmodels/MediaViewModel.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The LineageOS Project
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.lineageos.glimpse.viewmodels
+
+import androidx.lifecycle.*
+import androidx.lifecycle.ViewModelProvider.AndroidViewModelFactory.Companion.APPLICATION_KEY
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
+import org.lineageos.glimpse.GlimpseApplication
+import org.lineageos.glimpse.models.Album
+import org.lineageos.glimpse.models.Media
+import org.lineageos.glimpse.repository.MediaRepository
+
+class MediaViewModel(
+    private val savedStateHandle: SavedStateHandle, private val mediaRepository: MediaRepository
+) : ViewModel() {
+    private val _mediaPosition = savedStateHandle.getLiveData<Int>(MEDIA_POSITION_KEY)
+    val mediaPositionLiveData: LiveData<Int> = _mediaPosition
```

Comment:
types not needed

Reply:
this is "needed", mediaPositionInternal is MutableLiveData, I want to expose a non mutable variant